src/cpp/src/logit_processor.hpp

```diff
@@ -52,10 +82,26 @@ class TopKFilter : public ILogitTransformer {
 public:
     TopKFilter(size_t top_k) : m_top_k(top_k) {}
 
-    void apply(std::vector<Token>& logits) override {
-        std::sort(logits.begin(), logits.end(), [](const Token& lhs, const Token& rhs) {return lhs.m_log_prob > rhs.m_log_prob; });
-        size_t top_k = logits.size() >= m_top_k ? m_top_k : logits.size();
-        logits.resize(top_k);
+    // If this transform is used along with top_p, it should be applied after it since top_p sorts entire vector and top_k does it only partially
+    void apply(Logits& logits) override {
+        
+        /*
+        TODO: Uncommenting this section requires changes in reference texts in tests
+
+        if (m_top_k >= logits.m_size) 
```

Comment:
Please, uncomment and change refs in tests :)

Reply:
Agreed to do it in next PR